interactive: implement get_train_delays

multun opened the pull request.

multun committed:
interactive: implement get_train_delays

flomonster commented on src/main/java/fr/sncf/osrd/interactive/InteractiveSimulation.java:
```diff
@@ -145,4 +151,25 @@ public void watchChangesTypes(Set<ChangeType> changeTypes) throws IOException {
+
+    /** Sends back to the client the list of delays for all requested trains */
+    public void sendTrainDelays(Collection<String> trains) throws IOException {
+        if (hasUnexpectedState(SessionState.INITIALIZED))
```
`PAUSE` is also a valid state

flomonster reviewed (changes requested):
This won't work if we don't have a time table as input

multun committed:
interactive: rework state management

multun committed:
interactive: improve logging

multun committed:
interactive: implement the python side of get_train_delays

multun committed:
interactive: fix the server side of get_train_delays

When getting the delay at precisely the time of an event which doesn't move the train but still changes its state,
things failed because the TrainStateChange has no position update at all. In this case, get the position from
the current state.

multun committed:
fixup! interactive: fix the server side of get_train_delays
